File: Stage03-Locations/C#/Shared.cs
using System.Collections.Generic;

namespace Adventure_03_World
{
    /*
    The Items, Enemies and Locations are shared by multiple files, so are best kept in a code module.
    dictionaries of all these objects offer the greatest flexibility.

    Example of Items dictionary, which is a dictionary of item objects
    Items["torch"] = <Item object>
    Items["torch"].Name = "torch"
    Items["torch"].Description = "a flaming wooden torch"
    Items["key card"].Name = "key card",
    Items["key card"].Description = "Property of Holiday Inn"

    Example of Locations dictionary which is a dictionary of location objects
    Locations["hotel room"].Name = "hotel room"
    Locations["hotel room"].Description = "a damp hotel room"
    Locations["hotel room"].ToNorth = ""
    Locations["hotel room"].ToEast = "coridoor"
    Locations["hotel room"].ToSouth = ""
    Locations["hotel room"].ToWest = ""
    Locations["hotel room"].ItemRequired = "key card"              // key of an item from the items dictionary
    Locations["hotel room"].Items = {"torch","plastic sword"}      // a List of the dictionary keys of items in the location
    */

    internal static class Shared
    {
        public static bool Debug = true;
        public static Dictionary<string, int> Gamestates = new Dictionary<string, int> 
        {
            { "menu",       0},
            { "play",       1},
            { "leaderboard",2},
            { "quit",       3},
            { "dead",       4}
        };
        public static int Gamestate = 1;
        public static  Dictionary<string, Item> Items = new Dictionary<string, Item>();
        public static Dictionary<string, Location> Locations = new Dictionary<string, Location>();
        public static string CurrentLocation = "";
    }
}

File: Stage03-Locations/Python/shared.py
'''
the items, enemies and locations are shared by multiple files, so are best kept in a code module
dictionaries of all these objects offer the greatest flexibility

Example of items dictionary, which is a dictionary of item objects
items["torch"] = <item object>

items["torch"].name = "torch"
items["torch"].description = "a flaming wooden torch"
items["key card"].name" = "key card",
items["key card"].description = "Property of Holiday Inn"

example of locations dictionary which is a dictionary of location objects
locations["hotel room"].name = "hotel room"
locations["hotel room"].description = "a damp hotel room"
locations["hotel room"].to_north = ""
locations["hotel room"].to_east = "coridoor"
locations["hotel room"].to_south = ""
locations["hotel room"].to_west = ""
locations["hotel room"].item_required = "key card"              # key of an item from the items dictionary
locations["hotel room"].items = ["torch","plastic sword"]       # a Python list of the dictionary keys of items in the location
'''

debug = True
gamestates = {'menu':0, 'play':1, 'dead':2, 'quit':3}
gamestate = gamestates['menu']
items:dict = {}
locations:dict = {}
enemies:dict = {}
current_location:str = ""